### PR TITLE
Improve sd-gpg test

### PR DESCRIPTION
**Pull request by legoktm**
## Status

Ready for review

## Description of Changes

* Verify the fingerprints match the configured one
* Simplify regex to parse GPG output (--fingerprint made it harder!)
* Verify only one secret key is present in the keyring

## Testing

* [ ] CI passed
* [ ] Visual review

## Deployment

Any special considerations for deployment? n/a

## Checklist

- [x] All tests (`make test`) pass in `dom0`

### CODE REVIEW

File: tests/test_gpg.py [hunk 1]
```diff
@@ -10,8 +10,9 @@ class SD_GPG_Tests(SD_VM_Local_Test):
     def setUp(self):
         self.vm_name = "sd-gpg"
         super().setUp()
+        self.fingerprint = self.dom0_config["submission_key_fpr"]
```
Comment (cfm) [inline tests/test_gpg.py:13]:
```suggestion
        # Obtain the fingerprint explicitly configured in dom0 and injected into VMs.
        self.fingerprint = self.dom0_config["submission_key_fpr"]
```

File: tests/test_gpg.py [hunk 2]
```diff
@@ -10,8 +10,9 @@ class SD_GPG_Tests(SD_VM_Local_Test):
...
 
-    def getLocalFingerprint(self):
+    def get_dom0_fingerprint(self):
         """
         Obtain fingerprint for the key configured in dom0
```
Comment (cfm) [inline tests/test_gpg.py:17]:
```suggestion
        Obtain the fingerprint of the key actually present in dom0.
```

File: tests/test_gpg.py [hunk 3]
```diff
@@ -23,58 +24,38 @@ def getLocalFingerprint(self):
...
-            local_results = subprocess.check_output(
-                ["gpg", "-k", "--with-fingerprint"], env=gpg_env
-            )
+            local_results = subprocess.check_output(["gpg", "--list-secret-keys"], env=gpg_env)
             return self._extract_fingerprints(local_results)
 
-    def getRemoteFingerprints(self):
+    def get_vm_fingerprint(self):
         """
         Obtain fingerprints for all keys configured in GPG VM
```
Comment (cfm) [inline tests/test_gpg.py:32]:
```suggestion
        Obtain fingerprints for all keys actually present in GPG VM
```

File: tests/test_gpg.py [hunk 4]
```diff
@@ -23,58 +24,38 @@ def getLocalFingerprint(self):
...
-
-        # Exactly one fingerprint extracted from sd-journalist.sec
-        self.assertEqual(len(local_fp), 1)
-
-        # Local fingerprint is not falsy (e.g., "", None)
-        self.assertTrue(local_fp[0])
-
-        # At least one key in GPG VM keyring
-        self.assertGreater(len(remote_fp), 0)
+        """Verify the key in dom0 and sd-gpg match config.json"""
```
Comment (cfm) [inline tests/test_gpg.py:52]:
```suggestion
        """Verify the key present in dom0 and sd-gpg matches what's configured in config.json"""
```

**Review by cfm — CHANGES_REQUESTED** [review]
The revised test looks great.  I've left some nits to clarify some inline documentation that I found clouded rather than explained the test logic.  The distinction that I think is worthwhile is between the *fingerprint configured* somewhere versus the *key present* somewhere.

**Comment by legoktm** [comment]
Thanks, I've squashed in all your documentation suggestions.

**Review by cfm — APPROVED** [review]
Thanks, @legoktm.

**Pull request merged**